### PR TITLE
JP-2245 Implement the MAST AUI interface to the Engineering Database

### LINKED ISSUE #6274: Update ENGDB_Service to use the public AUI site
_Issue [JP-2245](https://jira.stsci.edu/browse/JP-2245) was created on JIRA by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham):_

Direct access to the engineering database will cease to exist with B7.8.1. The public is expect to [use MAST](https://mast.stsci.edu/portal_jwst/Mashup/Clients/JwstEdb/JwstEdb.html) for retrieving parameters. 

jwst.lib.engdb_tools.ENGDB_Service needs to be updated to allow use of this interface. The interface is currently not completely documented. See attached email for the API. There will probably be further updates before and while this issue is in action.

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Closes #6274 

Resolves [JP-2245](https://jira.stsci.edu/browse/JP-2245)

**Description**

This PR adds the MAST AUI interface to the JWST Engineering Database.

Checklist
- [x] Tests
- [x] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

**Review by hbushouse — APPROVED** [review]
Looks reasonable to me. It's the direct connection that's going to be going away in the future, right? Leaving only the MAST interface?

File: jwst/lib/engdb_direct.py [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+]
+
+
+class EngdbDirect(EngdbABC):
+    """
+    Access the JWST Engineering Database through direct connection
+
+    Parameters
+    ----------
+    base_url: str
```
[the remark this answers is not in the document]
Comment (stscieisenhamer) [reply jwst/lib/engdb_direct.py]: ✅

**Comment by stscieisenhamer** [comment]
WRT the different databases: Both interfaces will always be available. The direct one is necessary for OPS. The MAST is needed for usage outside of OPS.

File: CHANGES.rst [hunk 2]
```diff
@@ -51,6 +51,11 @@ jump
 
 - Updated jump detection step to use common code moved to stcal [#6089]
 
+lib
+---
+
+- Implement the MAST AUI interface to the Engineering Database [#6288]
```
Comment (nden) [inline CHANGES.rst:57]:
```suggestion
- Implement the MAST AUI interface to the Engineering Database. [#6288]
```

Comment (stscieisenhamer) [reply CHANGES.rst]: ✅

**Comment by stscieisenhamer** [comment]
Reviewers: A new documentation section, Library Utilities, has been added. This section includes all the significant docstrings for both the engineering database tools and set_telescope_pointing. The documentation itself is completely generated from docstrings, so module reviews will touch the content. However, note how the content is organized in the the documentation starting with `docs/jwst/lib/index.rst`

**Comment by stscieisenhamer** [comment]
woah cool. just realized that the documentation is built as part of CI. Click on "Details"  next to "docs/readthedocs.org" build line in the "All checks have passed" section.

File: CHANGES.rst (hunk 2 above)
Comment (stscieisenhamer) [reply CHANGES.rst]: ✅

**Pull request merged**